### DISCUSSION TITLE
Understanding the process for creating blazegraph

### DISCUSSION THREAD

**Discussion by shpoudel** [GRIDAPPSD/gridappsd-forum#103] [Q&A] [answered]
Hello Team,

When I select, usehouses=True in running the simulation are the houses inserted into the database on the fly or there are already houses in the database for the given model and the platform just runs that query to populate the GridLAB-D file.

I believe the second one is true. If that is the case, when are those houses added to the blazegraph database? I know there are python scripts in the Powergridmodel repo to do that, but I am not clear about the overall process.  Once the CIM XML file is loaded, the python scripts are then executed to update the database?

@poorva1209 @tonya1

**Comment by poorva1209** [comment]
Model xml is uploaded in the blazegraph and then scripts on the Powergrid-models repo are used to retrieve all the objects that will require measurements. Based on the objects, measurements are added in blazegraph using insert_measurement script. For more details see readme at https://github.com/GRIDAPPSD/Powergrid-Models/tree/develop/Meas. 

Let us know if you need more details.

**Comment by poorva1209** [comment, marked as answer]
The process for inserting houses is similar to measurements. Here is the readme link for that: https://github.com/GRIDAPPSD/Powergrid-Models/tree/develop/houses . After the CIM XML file is uploaded, the python scripts are then executed to update the database with measurements and houses.

**Reply by shpoudel** [reply]
Thanks, Poorva! That all makes sense to me now. I am asking these questions because I want to do something locally for OCHRE integration while Tom is working on a long-term solution.
